File: WebAPI/FinAnalyzer.Domain/Entities/Account.cs
namespace FinAnalyzer.Domain.Entities;

public class Account : BaseEntity
{
    public string Name { get; set; }

    public decimal Balance { get; set; }

    public AccountType AccountType { get; set; }
    public int AccountTypeId { get; set; }

    public Room Room { get; set; }
    public int RoomId { get; set; }

    public Person? Person { get; set; }
    public int? PersonId { get; set; }
}

File: mobile/app/api/interfaces/account.ts
export interface IAccount {
  id: number;
  name: string;
  balance: number;
  accountType: IAccountType;
  roomId: number;
  personName?: string;
}

export interface IAccountCreate {
  name: string;
  accountType: number;
}

export interface IAccountType {
  id: number;
  title: string;
}
